src/Octoshift/Handlers/DownloadLogsCommandHandler.cs

```diff
@@ -75,20 +73,17 @@ public async Task Handle(DownloadLogsCommandArgs args)
             _log.LogWarning($"Overwriting {args.MigrationLogFile} due to --overwrite option.");
         }
 
-        var result = await _retryPolicy.RetryOnResult(async () => await _githubApi.GetMigrationLogUrl(args.GithubOrg, args.GithubRepo), string.Empty,
+        var result = await _retryPolicy.RetryOnResult<(string MigrationLogURL, string MigrationID)>(async () => await _githubApi.GetMigrationLogUrl(args.GithubOrg, args.GithubRepo), result => string.IsNullOrEmpty(result.MigrationLogURL),
```

Comment:
I wouldn't change this retry logic.

Reply:
I think we have to make this change because the return value of the API method has changed. The filter has to adapt to that - unless I'm missing something.

Reply:
this retry is for when the log is yet unavailable and the method should still return `null` in that case so the retry logic  still applies. If we feel like we need to do refactors, it can be done in a separate PR.

Reply:
Okay! I understand this now - we can change `GetMigrationLogUrl` to continue to return `null`, and confusingly C#'s `string.Empty` matches empty strings _and_  nulls.